os/services/shell/shell-commands.c

```diff
@@ -734,7 +735,39 @@ shell_commands_init(void)
                                     echo_reply_handler);
 }
 /*---------------------------------------------------------------------------*/
-struct shell_command_t shell_commands[] = {
+void
+shell_command_set_register(struct shell_command_set_t *set)
+{
+  set->next = shell_command_sets_list;
```

Comment:
a sanity check for non-NULL pointer would be nice here

Reply:
What should the code be doing on NULL?

Reply:
I think once you switch to the list lib this will become immaterial?

Reply:
Right

Reply:
Umm. How does using list_push() make NULL immaterial?